src/SeriesProvider/SeriesProvider.py

```
@@ -105,38 +90,89 @@ def request_output(self, semaphoreSeriesDescription: SemaphoreSeriesDescription,
+        :param DBSeries: Series - A series of results to validate from the DB
+        :param DISeries: Series - A series of results to validate from data ingestion
+        :return a validated series: Series
 
-    def __merge_results(self, first: list, second: list) -> list:
-        """ Merges two lists together, will only keep unique results.
-            :param first: list[any] - The first list.
-            :param second: list[any] - The second list.
-            :return list: The combined lists.
         """
+        missing_results = 0
+        datetimeDict = self.__generate_datetime_map(timeDescription) # A dict with every time stamp we expect, with a value of none
 
-        #TODO:: This is a very slow, it was optimized with hashing like this:    
-            # uniqueToSecond = set(second) - set(first)
-            # return first + list(uniqueToSecond)
-        #But its not hashing properly anymore, needs to be looked into
+        # Iterate over every timestamp we are expecting 
+        for datetime in datetimeDict.keys():
+            
+            # Search the DB results for every input that matches that timestamp
+            valid_database_results = [input for input in DBSeries.data if input.timeGenerated == datetime]
+            if len(valid_database_results) > 1:
+                log(f'Warning: DB results returned multiple results for the same datetime \n{DBSeries}\n{valid_database_results}')
+
+            # Search the DI results for every input that matches that timestamp if there are DI results
+            if DISeries != None:
+                valid_ingestion_results = [input for input in DISeries.data if input.timeGenerated == datetime]
+                if len(valid_ingestion_results) > 1:
+                    log(f'Warning: DataIngestion results returned multiple results for the same datetime \n{DISeries}\n{valid_ingestion_results}')
```

is this a valid scenario?  if not, we should throw an error

---

Solved by switching to dicts